include/maliput/routing/route_phase.h

```diff
@@ -0,0 +1,175 @@
+// BSD 3-Clause License
+//
+// Copyright (c) 2023, Woven by Toyota. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice, this
+//   list of conditions and the following disclaimer.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// * Neither the name of the copyright holder nor the names of its
+//   contributors may be used to endorse or promote products derived from
+//   this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#pragma once
+
+#include <vector>
+
+#include "maliput/api/lane_data.h"
+#include "maliput/api/regions.h"
+#include "maliput/api/road_network.h"
+#include "maliput/common/maliput_copyable.h"
+#include "maliput/common/maliput_throw.h"
+#include "maliput/routing/route_position_result.h"
+
+namespace maliput {
+namespace routing {
+
+/// Manages a phase in a complete Route, towards its completion of a Route.
+///
+/// It is composed of a default maliput::api::LaneSRange which is the corridor
+/// the agent traversing the Route should follow. Alternative parallel adjacent
+/// LaneSRanges might be taken when provided.
```

Comment:
Does this go out to include more than the directly left/right lane?  e.g. four lane highway with on-ramp and exit 1km apart and just one segment connecting them (no branch points in the middle of the highway.

```
  ------x------A>-------x------
  ------x------B>-------x------
  ------x------C>-------x------
  ------x------D>-------x------
  --E--/                       \----F ---
```
Would it only include C & D?